### PR TITLE
MMT-2237 Updating to UMM-S v1.3.0

### PR DESCRIPTION
**Pull request by ryanmiller-1**
Overview of changes to the schema:
Added top level fields: Version Description, LastUpdatedDate, URL
Removed top level fields: Platforms, Related URLs, Science Keywords
Modified UseConstraints to have an additional field.
Removed ContactX fields from Service Organizations and added OnlineResource fields
Added SupportedReformattings.
Added/Modified valid values to some enums
Removed control from ProjectionAuthority

Associated changes to code:
Updated form schema for above changes
Updated datepicker to be usable in the form generator
Added new class to render datepickers in form generator
Added mechanism to override the description text from the schema with another description text from the schema in order to display accurate descriptions when elements are reused.
Updated JS to include urls and subfields in many of the behaviors of related urls on the forms. 
Updated JS to properly assign data levels (required fields) to Var/Service forms
Wrote data migration based on CMR's data migration
Updated and removed tests as necessary
Removed tests deemed to be redundant (form navigation tests checking saving the form, which is tested elsewhere).

### CODE REVIEW

File: app/helpers/drafts_helper.rb [hunk 1]
```diff
@@ -543,15 +526,10 @@ module DraftsHelper
   URLContentTypeMap = url_content_type_map.deep_dup
   # UMM-S v1.1 added TOOL to GET SERVICE subtypes
   umm_s_url_content_type_map = url_content_type_map_for_umm_s.deep_dup
-  umm_s_url_content_type_map['DistributionURL']['types']['GET SERVICE']['subtypes'] << ['Tool','TOOL']
-  umm_s_url_content_type_map['DistributionURL']['types']['GET SERVICE']['subtypes'].sort!
   UMMSURLContentTypeMap = umm_s_url_content_type_map
   URLContentTypeOptions = [
     ['Collection URL', 'CollectionURL'],
     ['Publication URL', 'PublicationURL'],
-    ['Data Center URL', 'DataCenterURL'],
```
Comment (ryanmiller-1) [inline app/helpers/drafts_helper.rb:552]: This change will be reverted and a new constant created to capture the UMM-S list created in the next commit

File: app/assets/javascripts/drafts_help.coffee [hunk 2]
```diff
@@ -62,6 +63,7 @@ $(document).ready ->
       when 'UOMLabel' then 'UOM Label'
       when 'AvgCompressionRateASCII' then 'Avg Compression Rate ASCII'
       when 'AvgCompressionRateNetCDF4' then 'Avg Compression Rate NetCDF4'
+      when 'URL Value' then 'URL Value'
```
Comment (ryanmiller-1) [inline app/assets/javascripts/drafts_help.coffee:66]: revisit

File: lib/assets/schemas/services/umm-s-form.json [hunk 3]
```diff
@@ -813,7 +855,7 @@
           "label": "Contact Groups",
           "help": "ContactGroups",
           "multiType": "ContactGroups",
-          "key": "/ContactGroups/items/properties/index_id",
+          "key": "ContactGroups/items/properties/index_id",
```
Comment (ryanmiller-1) [inline lib/assets/schemas/services/umm-s-form.json:858]: add comment to rehome these in 1.3.1

File: db/schema.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_03_02_154436) do
+ActiveRecord::Schema.define(version: 2020_05_21_150645) do
```
Comment (jceaser) [inline db/schema.rb:13]: why is this file changing?

Comment (ryanmiller-1) [reply db/schema.rb:13]: Running db:migrate for the data migration causes the version to update in this file.

File: lib/json_schema_form/umm_json_form.rb [hunk 5]
```diff
@@ -120,6 +121,16 @@ def sanitize_form_input(input, form_id, current_value = {})
     input
   end
 
+  # TODO: Remove when UMM_S 1.3.2 is implemented.
```
Comment (ryanmiller-1) [inline lib/json_schema_form/umm_json_form.rb:124]: This is a hack to get us around the schema capitalization difference between UMM-T and UMM-S. I hope to remove this in sprint 4.

File: lib/json_schema_form/umm_form_elements/custom_date_time.rb [hunk 6]
```diff
@@ -1,13 +1,14 @@
 # :nodoc:
 class CustomDateTime < UmmFormElement
+  # Uses the datepicker.coffee to generate the same date picker as UMM-C forms
   def render_markup
-    properties = element_properties(schema_fragment)
+    props = element_properties(schema_fragment)
```
Comment (jceaser) [inline lib/json_schema_form/umm_form_elements/custom_date_time.rb:5]: did the name clash with something? Why did you go with less descriptive?

File: db/migrate/20200521150645_umm_s_1_3_data_migration.rb [hunk 7]
```diff
@@ -0,0 +1,217 @@
+class UmmS13DataMigration < ActiveRecord::Migration[5.2]
```
Comment (charleshuang80) [inline db/migrate/20200521150645_umm_s_1_3_data_migration.rb:1]: I'm not sure how much this matters, but I feel that the methods here could use a lot more comments. This is a confusing change and having more comments could give more context. That being said, migrations are usu a one time thing and isn't code that some future developer will need to modify, and if there's an issue I assume you're going to fix it.

File: lib/json_schema_form/umm_form_elements/custom_date_time.rb (hunk 6 above)
Comment (charleshuang80) [inline lib/json_schema_form/umm_form_elements/custom_date_time.rb:5]: I prefer the changes that I had in here :)

**Comment by ryanmiller-1** [comment]
It was much easier to make a thorough migration because Erich gave me the migration that CMR did as a reference.

**Pull request merged**